### ISSUE TITLE
Suggestion : Better Component detection and interactions

### ISSUE THREAD

**Issue by Shikifuyin** [MightyPirates/OpenComputers#3284]
So I was basically playing DDSS with all those insanely complicated recipes ...
Then I got to the point of automating the silly forestry carpenter ...
I haven't use OC for quite some time so I though I would go this way.
Basic idea was to have a transposer detect incoming items in an input chest which is hooked to
RS/AE2 crafter/interface for auto-crafting. Then I would have some code to detect the requested
recipe and from there use some redstone IO blocks to control item routing and send them to the
carpenter with the correct recipe ...
Then I thought I could detect carpenter recipes dynamically by having one of their output item in
their inventory and read that with transposers too ...
Then I figured ... How am I gonna pair the transposer/redstoneIO blocks that match one carpenter ?
Basically I can't, there is absolutely no way to do that !!!! I have to give up dynamic detection and
use stupid hard-coded addresses ... that is such a shame !
(Another way would be for the Redstone IO block to be able to query surrounding blocks ...)
(Another way would be to detect adjacent pairs of components but it requires a debug card to get
components coordinates ... I do not want that !)

FIX : Allow all/most block components to have a common interface for querying adjacent blocks and
when detecting other OC components, allow them to retrieve their addresses !
This should not be too hard to implement ... and it would make component detection and
interactions MUCH more interesting !

**Comment by payonel** [comment]
we essentially flatten the network of components before the machine sees these components. this has come up before and we've decided to not redo the networking. i appreciate  your suggestion however, and i understand the current model increases setup tedium
